tests/spread/integration/git/task.yaml

```diff
@@ -0,0 +1,157 @@
+summary: Integration tests for git
+
+environment:
+  EXTRA_SLICES: "" # No extra slices by default.
+  # test git_bins
+  RUN/bins: test_bins
+  SLICE/bins: git_bins
+  # test git_http-support
+  RUN/http_support: test_http_support
+  SLICE/http_support: git_http-support
+  EXTRA_SLICES/http_support: ca-certificates_data
+  # test git_ftp-support
+  RUN/ftp_support: test_ftp_support
+  SLICE/ftp_support: git_ftp-support
+  # test git_daemon
+  RUN/daemon_support: test_daemon_support
+  SLICE/daemon_support: git_daemon
+
+execute: |
+
+  git_rootfs="/"
+  function chroot_git(){
+    # execute git within chroot. We use -C to cd since
+    # we have no $SHELL to navigate with. 
+    repo="$(realpath -e --relative-to="$git_rootfs" "$PWD")"
+    chroot "$git_rootfs" /usr/bin/git -C "$repo" "$@"
+  }
+
+  function setup(){
```

Comment:
You can put the logic of this function in a `prepare:` field in task.yaml.
See https://github.com/canonical/spread?tab=readme-ov-file#preparing